core/src/main/kotlin/io/customer/sdk/core/di/SDKComponentExt.kt

```diff
@@ -9,8 +9,10 @@ import android.content.Context
 /**
  * Create and register an instance of AndroidSDKComponent with the provided context,
  * only if it is not already initialized.
+ * This function should be called from all entry points of the SDK to ensure that
+ * AndroidSDKComponent is initialized before accessing any of its dependencies.
  */
-fun SDKComponent.registerAndroidSDKComponent(
+fun SDKComponent.ensureAndroidComponent(
```

Comment:
```suggestion
fun SDKComponent.setupAndroidComponent(
```

since we are not ensuring but rather initializing or reusing ?

Reply:
Happy renaming this, but `setup` is a bit confusing for me. If `AndroidSDKComponent` is already initialized, it simply returns existing instance. However if it's not initialized, it initializes the component and then returns new instance, ensuring `AndroidSDKComponent` is always initialized after this call and created only once. 🤔

Reply:
Discussed internally, we want to stick with `setupAndroidComponent`